From a8fcf207556ed44fc9bafa2e3b451b977080a947 Mon Sep 17 00:00:00 2001
From: Alberto Murillo <alberto.murillo.silva@intel.com>
Date: Wed, 30 Mar 2016 17:42:30 -0600
Subject: [PATCH] Fix argument type for _sd_notify() on python3

socket.sendall() expects a byte-string as
argument but takes its from _sd_notify in
a string format causing an exception on python3

Closes-Bug: #1564150

Change-Id: I16de988739b9e37b8b2721fefd3c6be485e5a4c7
---
 oslo_service/systemd.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/oslo_service/systemd.py b/oslo_service/systemd.py
index b213d32..777fe51 100644
--- a/oslo_service/systemd.py
+++ b/oslo_service/systemd.py
@@ -53,7 +53,7 @@ def notify():
     For details see
     http://www.freedesktop.org/software/systemd/man/sd_notify.html
     """
-    _sd_notify(False, 'READY=1')
+    _sd_notify(False, b'READY=1')
 
 
 def notify_once():
@@ -64,7 +64,7 @@ def notify_once():
     This method removes the NOTIFY_SOCKET environment variable to ensure
     notification is sent only once.
     """
-    _sd_notify(True, 'READY=1')
+    _sd_notify(True, b'READY=1')
 
 
 def onready(notify_socket, timeout):
-- 
2.5.5

